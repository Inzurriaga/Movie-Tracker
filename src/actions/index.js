// what gets sent
export const getMovies = (movies) => ({
  type: 'GET_MOVIES',
  movies
});

export const addUser = (userInfo) => ({
  type: "UPDATE_USER",
  userInfo
})

export const getGenres = (genre) => ({
  type: "GET_GENRES",
  genre
})
