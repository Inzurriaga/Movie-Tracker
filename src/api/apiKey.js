export const apiKey = '3033b069b2d0b32039c7d85cfe10cfd3';
