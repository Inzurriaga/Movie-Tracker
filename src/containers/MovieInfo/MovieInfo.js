import React, { Component } from "react"

class MovieInfo extends Component {
    render = () => {
        return(
            <div>
                {this.props.movieInfo.title}
            </div>
        )
    }
}

export default MovieInfo;