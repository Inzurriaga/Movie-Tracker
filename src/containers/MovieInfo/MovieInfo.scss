$font-title: 'Rajdhani', sans-serif;
$font-copy: 'Lato', sans-serif;

.MovieInfo {
  width: 100%;
  display: flex;
  background-size: cover;
  background-position: top center;
  background-repeat: no-repeat;
  padding: 2em;
  box-sizing: border-box;
  position: relative;

  &::before {
    content: "";
    position: absolute;
    width: 100%;
    height: 100%;
    top: 0;
    left: 0;
    background: #181b2dd9;
    z-index: 9;
  }

  .MovieInfo-Poster {
    width: 25%;
    z-index: 10;

    img {
      width: 100%;
      border-radius: 5px;
      box-shadow: 0px 2px 10px 3px #0000006b;
    }

  }

  .MovieInfo-Copy {
    width: 75%;
    z-index: 10;
    padding: 0em 2em;
    text-align: left;

    .Info-Title {
      color: #fff;
      font-family: $font-title;

      h3 {
        margin: 0;
        line-height: 1;
        font-size: 3em;
        letter-spacing: .01em;
        font-weight: 400;
      }
    }

    .Info-Extras {
      display: flex;
      color:#fff;
      font-family: $font-title;

      button {
        background:none;
        border:none;
        outline: none;
        cursor: pointer;
        color:#fff;
      }
    }

    .Info-Copy {
      h5 {
        color:#fff;
        margin: 0;
        font-family: $font-copy;
      }
      p {
        margin: 0;
        font-family: $font-copy;
        color: #fff;
        font-weight: 100;
        letter-spacing: .03em;
      }
    }
    .MovieInfo-Trailer {
      z-index: 10;

      iframe {
        width: 100%;
      }
    }

  }




}
