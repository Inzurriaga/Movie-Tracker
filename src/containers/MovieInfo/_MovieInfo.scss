.MovieInfo {
  width: 100%;
  display: flex;
  background-size: cover;
  background-position: top center;
  background-repeat: no-repeat;
  padding: 2em;
  box-sizing: border-box;
  align-items: center;
  position: relative;
  // height: 93%;
  box-shadow: inset 0px 0px 18px 11px black;

  &::before {
    content: "";
    position: absolute;
    width: 100%;
    height: 100%;
    top: 0;
    left: 0;
    background: #755e0b6b;
    z-index: 9;
  }

  .MovieInfo-Poster {
    width: 35%;
    z-index: 11;
    box-sizing: border-box;
    position: relative;

    img {
      width: 100%;
      border-radius: 5px;
      box-shadow: 0px 2px 10px 3px #0000006b;
    }

    .Favorite_True {
      color: #dcdc26;
      background: #ffffff61;
    }

    .Favorite_False {
      color: #a2a2a2;
      background: #545454b0;
    }

    button {
      background: none;
      border-top-left-radius: 5px;
      border: none;
      outline: none;
      cursor: pointer;
      position: absolute;
      font-size: 3em;
      padding: 7px;
      line-height: 1.5;
      background: #545454b0;
      box-shadow: 2px 3px 11px 0px #231f2047;
      transition: all .5s ease;

      &:hover {
        color: #dcdc26;
        background: #ffffff9c;
      }
    }

  }

  .MovieInfo-Copy {
    width: 65%;
    padding: 1.25em 2em 1.25em 3em;
    text-align: left;
    box-sizing: border-box;
    background: #0c0f29a1;
    border-radius: 5px;
    position: relative;
    right: 1em;
    z-index: 10;

    .Info-Title {
      color: #fff;
      font-size: 2em;
      font-family: $font-title;
      padding-bottom: .5em;

      h3 {
        margin: 0;
        line-height: 1;
        font-size: 1.5em;
        letter-spacing: .01em;
        font-weight: 400;
        line-height: .9;

        span {
          font-size: .5em;
        }
      }
    }

    .Info-Copy {
      margin-bottom: 2em;

      h5 {
        color:#fff;
        margin: 0;
        font-family: $font-copy;
        font-size: 1.2em;
        padding-bottom: .3em;
        font-weight: 400;
        letter-spacing: .05em;

        .Info-Score {
          font-size: .8em;
          font-weight: 100;
        }
      }

      p {
        margin: 0;
        font-family: $font-copy;
        color: #fff;
        font-weight: 100;
        letter-spacing: .02em;
        font-size: 1.2em;
        line-height: 1.2;
      }

    }
    .MovieInfo-Trailer {
      z-index: 10;

      iframe {
        width: 100%;
        height: 340px;
        border: none;
        box-shadow: 0px 2px 10px 3px #0000006b;
      }
    }

  }

}
