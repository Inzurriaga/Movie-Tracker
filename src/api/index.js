import { apiKey } from './apiKey'

const backEndUrl = 'http://localhost:3000/api/'

// Movies - DB Fetch Calls
const fetchMovies = async (url, urlEnd) => {
  let movieUrl = `${url}${apiKey}${urlEnd}`;
    const response = await fetch(movieUrl);
    const movies = await response.json()
    return movies
}

// Users - Backend Fetch Calls
const settings = (method, body) => ({
  method,
  headers: {'Content-Type': 'application/json'},
  body: JSON.stringify(body)
})

const deleteSettings = () => ({
  method: 'DELETE',
  headers: {'Content-Type': 'application/json'}
})

const postFetch = async (url, method, body) => {
  const response = await fetch(`${backEndUrl}${url}`, settings(method, body))
  return response;
}

const getFetch = async (url) => {
  const response = await fetch(`${backEndUrl}${url}`)
  if(!response.ok) {
    return 'User Already Exists'
    // throw Error(response.statusText)
  } else {
    const user = await response.json()
    return user.data;
  }
    // if(response.status === 'success') {}
}

const deleteFetch = async (url) => {
  const response = await fetch(`${backEndUrl}${url}`, deleteSettings())
  const retrievedData = await response.json()
  return retrievedData
}





export { fetchMovies, postFetch, getFetch, deleteFetch, settings, deleteSettings }
