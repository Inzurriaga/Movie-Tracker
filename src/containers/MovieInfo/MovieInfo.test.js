import React from 'react';
import { MovieInfo, mapStateToProps, mapDispatchToProps } from './MovieInfo';
import { addFavorite, removeFavorite } from "../../actions/index"
import { shallow } from 'enzyme';
import { fetchMovies } from "../../api/index"
jest.mock("../../api/index")



const mockFavorite = ['123','1234','2341','1235'];
const mockUser = {}
const mockId = '123';
const mockCurrentMovie = [{
  backdrop_path: "/ADJ6V8W96It4KElY2SPZvkKPBR.jpg",
  vote_average: 7.4,
  id: 458723,
  poster_path: "/ux2dU1jQ2ACIMShzB3yP93",
  overview: "Husband and wife Gabe and",
  tagline: "Watch yourself",
  title: "Us",
  backdrop_path: "/ADJ6V8W96It4KElY2SP",
  release_date: "2019-03-14",
  videos: { results: [{id: "5c2238fc0e0a264df2ef7cd4"}] }
}]
const mockhandleFavorite = jest.fn();

describe('MovieInfo', () => {
  describe('MovieInfo Component', () => {
    let movieInfoCompenent;

    let mockhandleAdd = jest.fn();
    beforeEach(() => {
      movieInfoCompenent = shallow(<MovieInfo user={mockUser}
                                              favorites={mockFavorite}
                                              movieID={mockId}
                                              removeFavorite={mockhandleFavorite}
                                              addFavorite={mockhandleAdd}
                                            />)

    })

    it('should have default properties', () => {
      expect(movieInfoCompenent.state()).toEqual({
        favoriteStatus: false,
      });
    })

    it('should match the snapshot if there is a currentMovie', () => {
      movieInfoCompenent.setState({
        currentMovie: mockCurrentMovie,
      })
      expect(movieInfoCompenent).toMatchSnapshot()
    })

    it('should match the snapshot and show favorite button if a user exists', () => {
      movieInfoCompenent.setState({
        currentMovie: mockCurrentMovie,
      })
      movieInfoCompenent.setProps({ user: {id: 7, name: "a", password: "a", email: "a"}})
      expect(movieInfoCompenent).toMatchSnapshot()
    })

    it.skip('should on removeFavorite spy has been invoked', async () => {
      movieInfoCompenent.instance().removeFavorite('123');
      expect(mockhandleFavorite).toHaveBeenCalled()
    })

    it.skip('should on addFavorite spy has been invoked', async () => {
      movieInfoCompenent.instance().addFavorite();
      expect(mockhandleAdd).toHaveBeenCalled()
    })

    it.skip('should on addFavorite spy has been invoked', async () => {
      movieInfoCompenent.instance().handleFavorite('123');
      expect(movieInfoCompenent.instance().removeFavorite).toHaveBeenCalledWith('123')
    })

  })

  describe('mapStateToProps', () => {
    it('should return a object', () => {
      const mockState = {
        user: {},
        favorites: [{},{},{}],
        movies: []
      };
      const expected = {
        user: {},
        favorites: [{},{},{}]
      };
       const result = mapStateToProps(mockState);
       expect(result).toEqual(expected)
    })
  })

  describe('mapDispstchToProps', () => {
    it('should addFavorite been called with input argument', () => {
      const mockDispatch = jest.fn();
      const mockFavorite = 123;
      const actionToDispatch = addFavorite(mockFavorite);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.addFavorite(mockFavorite);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)

    })
    it('should removeFavorite been called with input argument ', () => {
      const mockDispatch = jest.fn();
      const mockFavorite = 123;
      const actionToDispatch = removeFavorite(mockFavorite);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.removeFavorite(mockFavorite);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })

})
