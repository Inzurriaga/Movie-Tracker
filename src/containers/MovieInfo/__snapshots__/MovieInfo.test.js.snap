// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieInfo MovieInfo Component should match the snapshot and show favorite button if a user exists 1`] = `
<div>
  Loading
</div>
`;

exports[`MovieInfo MovieInfo Component should match the snapshot if there is a currentMovie 1`] = `
<div>
  Loading
</div>
`;
